clients/client-go/tcobject/upload_download_test.go

```diff
@@ -0,0 +1,152 @@
+package tcobject_test
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/cenkalti/backoff/v3"
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+	"github.com/taskcluster/httpbackoff/v3"
+	"github.com/taskcluster/taskcluster/v42/clients/client-go/tcobject"
+	"github.com/taskcluster/taskcluster/v42/workers/generic-worker/mocktc"
```

Comment:
Let's move this package to `github.com/taskcluster/taskcluster/internal/mocktc` since it's now shared between this package and generic-worker.

Reply:
I agree. I tried this just now, and it created a lot of noise for this PR - is it ok for me to do that in a follow-up PR?

Reply:
Yeah, sounds good as long as it's not P5 :)

Reply:
Created #4696